### PR TITLE
Spiral array

### LINKED ISSUE #932: New algorithm for spiral array printing
@jainaman224  Can i add the code for the spiral array algorithm?

### PR DESCRIPTION
**Pull request by vishnu0179**
#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

New Algortihm Added: Spiral Array

#### Languages Used:

-C

#### Files Added:
- Spiral_Array.c 
- readme.md

### CODE REVIEW

**Comment by marwahmanbir** [comment]
- Fix checklist above. Instead of [o], check the box with [x]
- Rename readme.md to README.md

**Comment by vishnu0179** [comment]
This pull request does not resolve any issue instaed i added a new algortihm to the repository so what should i add in "Fixes#"??

**Comment by marwahmanbir** [comment]
@vishnu0179 You are supposed to first create an issue mentioning your proposed changes and if you are assigned by @jainaman224 then only you can work on it.

**Comment by vishnu0179** [comment]
I Have updated the fixes no. Please Check

File: Spiral_Array/Spiral_Array.c [hunk 1]
```diff
@@ -0,0 +1,71 @@
+#include <stdio.h>
+
+//SPIRAL PRINTING 2-D matrix
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Move this at the top, before header file.

File: Spiral_Array/Spiral_Array.c [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+int main()
+{
+    int m,n;
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Add a space after comma.

File: Spiral_Array/Spiral_Array.c [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    int left, right, top, bottom;
+    int i, j, count, dir;
+    int arr[100][100];
+    scanf("%d %d",&m,&n);  // Matrix of m*n
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Space after each comma.

File: Spiral_Array/Spiral_Array.c [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    left = 0;
+    right = n-1;
+    top = 0;
+    bottom = m-1;
+
+    count = m*n;        //no. of elements
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Space around ``*``

File: Spiral_Array/Spiral_Array.c [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    dir=1;
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Space around ``=``

File: Spiral_Array/Spiral_Array.c [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+    for(i=0; i<m ; i++)	// Taking Inputs
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Space after ``for``

File: Spiral_Array/Spiral_Array.c [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        for(j=0;j<n;j++)
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Space after ``for`` and after each semicolon.

File: Spiral_Array/Spiral_Array.c [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    top = 0;
+    bottom = m-1;
+
+    count = m*n;        //no. of elements
+    dir=1;
+
+    for(i=0; i<m ; i++)	// Taking Inputs
+    {
+        for(j=0;j<n;j++)	
+            scanf("%d",&arr[i][j]);
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Space after comma

File: Spiral_Array/Spiral_Array.c [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    while(left<=right&&top<=bottom&&count>0)
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Space around ``&&``.

File: Spiral_Array/Spiral_Array.c [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        if(dir==1)		//left to right
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Space after ``if`` and around ``==``

File: Spiral_Array/Spiral_Array.c [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            for(i = left; i<=right; i++)
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Fix spacing

File: Spiral_Array/Spiral_Array.c [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                printf("%d  ",arr[top][i]);	//Printing the topmost untraversed row
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Fix spacing

File: Spiral_Array/Spiral_Array.c [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            for(i = left; i<=right; i++)
+            {
+                printf("%d  ",arr[top][i]);	//Printing the topmost untraversed row  
+                count--;
+            }
+            dir++;
+            top++;
+        }
+        if(dir==2)		//top to bottom
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Fix spacing

File: Spiral_Array/Spiral_Array.c [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            for(i=top;i<=bottom;i++)
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Fix spacing

File: Spiral_Array/Spiral_Array.c [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                printf("%d  ",arr[i][right]);	//Printing the rightmost untraversed column
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Fix spacing

File: Spiral_Array/Spiral_Array.c [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+                count--; 
+            }
+            dir++;
+            right--;
+        }
+        if(dir==3)		//left to right
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Fix spacing

File: Spiral_Array/Spiral_Array.c [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            for(i= right; i>=left;i--)
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Fix spacing

File: Spiral_Array/Spiral_Array.c [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                printf("%d  ",arr[bottom][i]);	//Printing the bottommost untraversed column
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Fix spacing

File: Spiral_Array/Spiral_Array.c [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+                count--;
+            }
+            dir++;
+            bottom--;
+        }
+        if(dir==4)		//bottom to top
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Fix spacing

File: Spiral_Array/Spiral_Array.c [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            for(i = bottom; i>=top; i--)
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Fix spacing

File: Spiral_Array/Spiral_Array.c [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                printf("%d  ",arr[i][left]);	//Printing the leftmost untraversed column
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Fix spacing

File: Spiral_Array/Spiral_Array.c [hunk 22]
```diff
@@ -0,0 +1,71 @@
...
+                count--;
+            }
+            dir=1;
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: Fix spacing

File: Spiral_Array/Spiral_Array.c [hunk 23]
```diff
@@ -0,0 +1,71 @@
...
+            left++;
+        }
+    }
+    return 0;
+}
```
Comment (marwahmanbir) [inline Spiral_Array/Spiral_Array.c]: After line 71, add in order:
- Single newline
- Sample input-output
- Single newline

**Comment by marwahmanbir** [comment]
Spacing has been suggested to follow coding conventions (https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141388.html)

**Comment by vishnu0179** [comment]
@marwahmanbir  Done with the changes

**Comment by marwahmanbir** [comment]
Good work. :+1:

**Pull request merged**